auth/secp256k1/secp256k1.go

```diff
@@ -6,5 +6,6 @@ import (
 
 // Secp256k1Signer implements the Signer interface using secp256k1 keys
 type Secp256k1Signer struct {
+	//nolint:unused
```

Comment:
The correct fix here is to move this `type Secp256k1Signer` to `secp256k1_evm.go` and then add another `type Secp256k1Signer` to `secp256k1_no_evm` that doesn't have this `privateKey` field.